app/main.py

```diff
@@ -1,2 +1,96 @@
+from typing import Callable
+
+
 class Dictionary:
-    pass
+    def __init__(self) -> None:
+        self.capacity = 8
+        self.load_factor = 2 / 3
+        self.resize = 2
+        self.length = 0
+        self.hash_table = [None] * self.capacity
+
+    def __setitem__(self, key: Callable, value: Callable) -> None:
+        index = hash(key) % self.capacity
+        while (self.hash_table[index] is not None
+               or self.hash_table[index] == 1):
+            if self.hash_table[index] == 1 or self.hash_table[index][0] == key:
```

Comment:
also, compare hashes first - this operation performs faster because it's comparing not complex keys (tuples) but integers

Reply:
I didn’t quite understand what I should compare earlier - the conditions in the if or something else?